networkx/algorithms/tests/test_cycles.py

```diff
@@ -863,3 +866,29 @@ def test_complete_graph(self):
     def test_tree_graph(self):
         tg = nx.balanced_tree(3, 3)
         assert not minimum_cycle_basis(tg)
+
+
+class TestGirth:
+    def test_chvatal_graph(self):
+        graph = nx.chvatal_graph()
+        assert girth(graph) == 4
+
+    def test_tutte_graph(self):
+        graph = nx.tutte_graph()
+        assert girth(graph) == 4
+
+    def test_petersen_graph(self):
+        graph = nx.petersen_graph()
+        assert girth(graph) == 5
+
+    def test_heawood_graph(self):
+        graph = nx.heawood_graph()
+        assert girth(graph) == 6
+
+    def test_pappus_graph(self):
+        graph = nx.pappus_graph()
+        assert girth(graph) == 6
+
+    def test_random_tree(self, n=100):
+        graph = nx.random_tree(n, seed=42)
+        assert girth(graph) == inf
```

Comment:
I'd vote to parametrize this rather than use classes:

```suggestion
@pytest.mark.parametrize(
    ("G", "expected"),
    (
        (nx.chvatal_graph(), 4),
        (nx.tutte_graph(), 4),
        (nx.petersen_graph(), 5),
        (nx.heawood_graph(), 6),
        (nx.pappus_graph(), 6),
        (nx.random_tree(10, seed=42), inf),
    ),
)
def test_girth(G, expected):
    assert nx.girth(G) == expected
```

Reply:
Yes, this is much more concise. I must confess I'm not used to testing code the way it's been implemented in these files, so I just complied with the previous examples.